modules/loader-utils/src/lib/file-provider/range-request-file.ts

```diff
@@ -0,0 +1,137 @@
+// loaders.gl
+// SPDX-License-Identifier: MIT
+// Copyright (c) vis.gl contributors
+import {FileProvider} from './file-provider';
+
+/**
+ * Provides file data using range requests to the server
+ * @deprecated - will be replaced with ReadableFile
+ */
+export class RangeRequestFile implements FileProvider {
+  /** The File object from which data is provided */
+  private url: string;
+  private size: bigint;
+
+  /** Create a new BrowserFile */
+  private constructor(url: string, size: bigint) {
+    this.url = url;
+    this.size = size;
+  }
+
+  static async create(url: string) {
+    const res = await fetch(url, {method: 'HEAD'});
+    const size = parseInt(res.headers.get('content-length') ?? '');
+    return new RangeRequestFile(url, BigInt(size));
+  }
+
+  /**
+   * returns an ArrayBuffer whose contents are a copy of this file bytes from start, inclusive, with provided length.
+   * @param start The offset, in byte, from the start of the file where to start reading the data.
+   * @param lenght Length of read data
+   */
+  private async getBytesFromFile(start: number, lenght: number): Promise<ArrayBuffer> {
+    const res = await fetch(this.url, {
```

Comment:
My worry with FileProvider is that makes lots of tiny reads - that will absolutely kill performance for anything network based.